cors/cors.go

```diff
@@ -0,0 +1,127 @@
+package cors
+
+import (
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/events"
+)
+
+type OriginChecker interface {
+	Options(events.APIGatewayV2HTTPRequest) events.APIGatewayV2HTTPResponse
+	GetCorsHeaders(request events.APIGatewayV2HTTPRequest) (map[string]string, bool)
+}
+
+// Domains contains the allowed domains and the accepted methods for each one
+type Domains struct {
+	Allowed map[string]map[string]bool
+}
+
+// NewDomains returns the default Domains
+func NewOriginChecker() OriginChecker {
+	acceptedDomains := map[string]map[string]bool{
+		"thelist.app": {"DELETE": true, "GET": true, "PATCH": true, "POST": true},
```

Comment:
I worry that we've made things a bit over-engineered, do we expect to allow different methods for different hostnames? Won't it always be the same?

Reply:
This is likely - I may have got carried away